libc/CMakeLists.txt

```diff
@@ -54,27 +54,12 @@ set(LIBC_NAMESPACE ${default_namespace}
 add_subdirectory(newhdrgen)
 
 
-if(LLVM_LIBC_FULL_BUILD OR LLVM_LIBC_GPU_BUILD)
-  if(NOT LIBC_HDRGEN_EXE)
-    # We need to set up hdrgen first since other targets depend on it.
-    add_subdirectory(utils/LibcTableGenUtil)
-    add_subdirectory(utils/HdrGen)
-    # Calling add_tablegen sets variables like LIBC_TABLEGEN_EXE in
-    # PARENT_SCOPE which get lost until saved in the cache.
-    set(LIBC_TABLEGEN_EXE "${LIBC_TABLEGEN_EXE}" CACHE INTERNAL "")
-    set(LIBC_TABLEGEN_TARGET "${LIBC_TABLEGEN_TARGET}" CACHE INTERNAL "")
-  else()
-    message(STATUS "Will use ${LIBC_HDRGEN_EXE} for libc header generation.")
-  endif()
-endif()
 # We will build the GPU utilities if we are not doing a runtimes build.
 option(LIBC_BUILD_GPU_LOADER "Always build the GPU loader utilities" OFF)
 if(LIBC_BUILD_GPU_LOADER OR (LLVM_LIBC_GPU_BUILD AND NOT LLVM_RUNTIMES_BUILD))
   add_subdirectory(utils/gpu)
 endif()
 
-option(LIBC_USE_NEW_HEADER_GEN "Generate header files using new headergen instead of the old one" ON)
-
 set(NEED_LIBC_HDRGEN FALSE)
```

Comment:
This can be deleted?
```suggestion
```

Reply:
Unsure.  Let me check the original context under which it was added.
9dbedcac1243e8e99103bdff37da51dded67b766

Reply:
also related c4f236e94a4e2df46838a540936bb9b61c771ca1

Reply:
FYI the GPU build still uses that logic to make the `amdhsa-loader` and `nvptx-loader` utilities.

Reply:
right, I think I left the important bits intact in 46251c628e4e4aee38f15419fcc67cc8b6c4d827, PTAL.